nucleus/payara-modules/payara-executor-service/src/main/java/fish/payara/nucleus/executorservice/PayaraExecutorService.java

```diff
@@ -120,8 +120,6 @@ public void postConstruct() {
         if (transactions != null) {
             transactions.addListenerForType(PayaraExecutorServiceConfiguration.class, this);
         }
-
-        initialiseThreadPools();
```

Comment:
I believe it is this point that makes the tests fail. Reinitializing at ALL_APPLICATIONS_LOADED was a heuristic I found to work for multiple embedded starts, but the threadpool is already needed during load.

Reply:
Seems reasonable. I have reworked it slightly to follow the same strategy but use null value of pool as the trigger to recreate